### PR TITLE
Enable plugins also in upstream

### LINKED ISSUE #391: Upstream installer does not install plugins which are included in Satellite
According to our upstream automation and `standalone installer` jobs, installed katello (satellite-upstream) doesn't have plugins installed (discovery, libvirt, etc.) which causes our tests to fail.

I believe we consider `upstream` Katello with all plugins which are bundled together and installed by default in Satellite 6 therefore I'm marking this issue as a bug

### PR DESCRIPTION
**Pull request by lpramuk**
Enable plugins also in upstream
- setup_default_capsule() is run also for upstream and zstream distribution
- product_install(): dont exlude upstream distro from postinstall plugins enablements
- upstream_install(): some pkgs needs to be installed explicitly in upstream

(Fixes #391)

### CODE REVIEW

**Comment by rplevka** [comment]
ACK

File: automation_tools/__init__.py [hunk 1]
```diff
@@ -1322,18 +1322,17 @@ def product_install(distribution, create_vm=False, certificate_url=None,
...
+            execute(setup_libvirt_key, host=host)
+        if satellite_version != '6.0':
+            execute(install_puppet_scap_client, host=host)
+        if satellite_version == '6.1':
+            execute(setup_oscap, host=host)
+        if satellite_version in ('6.2', ''):
+            execute(oscap_content, host=host)
+        if os.environ.get('PXE_DEFAULT_TEMPLATE_URL') is not None:
+            execute(setup_foreman_discovery, host=host)
+        execute(enable_ostree, host=host)
```
Comment (kbidarkar) [inline automation_tools/__init__.py]: Let's not change this unless we, have explicitly added support for  upstream.
There are features, which have it's own requirements, it would be better if the feature owner adds support for it in the upstream.

Example: oscap installs a package in downstream , but needs a puppet-module in upstream.

Comment (lpramuk) [inline automation_tools/__init__.py]: - satellite_version in ('6.2', '') - it doenst mean upstream it will be skipped anyway, I can add explicit condition to skip in upstream only this. I want the rest merged soon.


- default smart proxy isn't nice bonus?

Comment (lpramuk) [inline automation_tools/__init__.py]: Kedar, I will refactor it to meet your comments. thanks

**Comment by lpramuk** [comment]
Final state of plugins/tasks for upstream: 
- setup_default_capsule enabled
- setup_libvirt_key enabled
- setup_foreman_discovery enabled
- ostree still disabled (will handle lpramuk in new PR)
- oscap still disabled (will handle kbidarka in new PR)

**Comment by lpramuk** [comment]
cloned satellite6-installer pointed at my branch was successful installing upstream:
```
[hostname] Executing task 'setup_default_capsule'
...
[hostname] Executing task 'setup_libvirt_key'
...
[hostname] Executing task 'setup_foreman_discovery'
...
Done.
Disconnecting from root@hostname... done.
+ '[' false = true ']'
Deleting 6 temporary files
Started calculate disk usage of build
Finished Calculation of disk usage of build in 0 seconds
Started calculate disk usage of workspace
Finished Calculation of disk usage of workspace in 0 seconds
Finished: SUCCESS
```

**Comment by rplevka** [comment]
ACK
looks good to me

File: automation_tools/__init__.py [hunk 2]
```diff
@@ -1322,18 +1322,18 @@ def product_install(distribution, create_vm=False, certificate_url=None,
     if distribution.startswith('satellite6'):
         execute(setup_default_docker, host=host)
         execute(katello_service, 'restart', host=host)
+        if os.environ.get('LIBVIRT_KEY_URL') is not None:
```
Comment (omaciel) [inline automation_tools/__init__.py]: Can we add a comment explaining the logic here? 3 months from now we want to remember why we have this condition :)

File: automation_tools/__init__.py [hunk 3]
```diff
@@ -1322,18 +1322,18 @@ def product_install(distribution, create_vm=False, certificate_url=None,
...
             if satellite_version != '6.0':
                 execute(install_puppet_scap_client, host=host)
             if satellite_version == '6.1':
                 execute(setup_oscap, host=host)
             if satellite_version in ('6.2', ''):
                 execute(oscap_content, host=host)
-            if os.environ.get('PXE_DEFAULT_TEMPLATE_URL') is not None:
-                execute(setup_foreman_discovery, host=host)
             execute(enable_ostree, host=host)
+        if os.environ.get('PXE_DEFAULT_TEMPLATE_URL') is not None:
```
Comment (omaciel) [inline automation_tools/__init__.py]: Add a comment?

Comment (lpramuk) [inline automation_tools/__init__.py]: coz the task that is called uses that env var... ugly

we are going to refactor so that tasks will have params instead.

Comment (lpramuk) [inline automation_tools/__init__.py]: I'm going to add some explanation..er..comment

File: automation_tools/__init__.py [hunk 4]
```diff
@@ -961,6 +961,8 @@ def upstream_install(
             os_version=os_version,
             sam='--sam' if sam else ''
         ))
+    run('yum install -y '
```
Comment (omaciel) [inline automation_tools/__init__.py]: This works but I was wondering if we should create a PLUGINS constant and list them there, in alphabetical order?

```python
PLUGINS = [
    'ec2',
    'gce', 
    'libvirt', 
    'openstack', 
    'ovirt', 
    'rackspace', 
    'vmware',
]

    run('yum install -y '
           'foreman-{{{0}}}'.format(','.join(PLUGINS))
```

**Comment by omaciel** [comment]
@lpramuk none of my comments are blockers and can be either taken care of here or in a subsequent PR. ACK

File: automation_tools/__init__.py (hunk 4 above)
Comment (lpramuk) [inline automation_tools/__init__.py]: I was thinking of it too as I saw it elsewhere in the code :)

**Comment by omaciel** [comment]
Will come back later and will merge it as soon as @lpramuk adds one more change (he told me via irc that he would also handle my comment about plugins).

**Comment by lpramuk** [comment]
Comments addressed.
- by adding comments
- by adding tabular list of compute resources to be installed for upstream

File: automation_tools/__init__.py [hunk 5]
```diff
@@ -1251,29 +1263,27 @@ def product_install(distribution, create_vm=False, certificate_url=None,
...
-                interface=os.environ.get('INTERFACE') or 'eth0',
-                run_katello_installer=False
-            )[host])
-        else:
-            # execute returns a dictionary mapping host strings to the given
-            # task's return value.
-            installer_options.update(execute(
-                setup_default_capsule, host=host, run_katello_installer=False
-            )[host])
+    # When using VLAN Bridges os.environ.get('BRIDGE') is 'true' and
```
Comment (omaciel) [inline automation_tools/__init__.py]: Forgot to ask but why did you remove this conditional?

```diff
-    if distribution in (
 -            'satellite6-cdn', 'satellite6-downstream', 'satellite6-iso'):
```

Comment (kbidarkar) [inline automation_tools/__init__.py]: This is removed because we now want to configure, setup_default_capsule() for upstream, which was not being configured earlier.

**Comment by kbidarkar** [comment]
ACK, all the suggestions and comments addressed.

File: automation_tools/__init__.py [hunk 6]
```diff
@@ -962,6 +962,18 @@ def upstream_install(
...
+    compute_resources = [
+        'ec2',
+        'gce',
+        'libvirt',
+        'openstack',
+        'ovirt',
+        'rackspace',
+        'vmware',
+    ]
+    run('yum install -y foreman-{{{0}}}'.format(','.join(compute_resources)))
```
Comment (sghai) [inline automation_tools/__init__.py]: :+1:
